### PR TITLE
Support HTTP POST request in the htmlparser and regex source

### PR DESCRIPTION
**Pull request by yuyichao**
Example:

    source = "regex"
    regex = "spcm_linux_libs_v[0-9a-zA-Z]*"
    url = "https://spectrum-instrumentation.com/spcm_downloads_downloads_ajax"
    post_data = "series%5B%5D=273&families%5B%5D=475"

### CODE REVIEW

File: nvchecker/httpclient/aiohttp_httpclient.py [hunk 1]
```diff
@@ -47,7 +48,13 @@ async def request_impl(
 
     if proxy is not None:
       kwargs['proxy'] = proxy
-    if json is not None:
+    if body is not None:
+      # Make sure all backends have the same default encoding for post data.
+      if 'Content-Type' not in headers:
+        headers = {**headers, 'Content-Type': 'application/x-www-form-urlencoded'}
```
Comment (lilydjwg) [inline nvchecker/httpclient/aiohttp_httpclient.py:54]: Why not use `headers.setdefault(...)`?

**Review by lilydjwg — COMMENTED** [review]
Please update tests and docs.

File: nvchecker/httpclient/aiohttp_httpclient.py (hunk 1 above)
Comment (yuyichao) [reply nvchecker/httpclient/aiohttp_httpclient.py:54]: To not mutate the input.

Comment (lilydjwg) [reply nvchecker/httpclient/aiohttp_httpclient.py:54]: I see.

**Comment by yuyichao** [comment]
Tests added and document updated.

I did not add a test for the htmlparser source since I couldn't figure out how to let httpbin reply a html to a post request...

**Pull request merged**